rtloader/common/builtins/_util.c

```diff
@@ -162,6 +172,62 @@ PyObject *subprocess_output(PyObject *self, PyObject *args)
         subprocess_args[i] = subprocess_arg;
     }
 
+    if (cmd_env != NULL && cmd_env != Py_None) {
+        if (!PyDict_Check(cmd_env)) {
+            PyErr_SetString(PyExc_TypeError, "env is not a dict");
+            goto cleanup;
+        }
+
+        subprocess_env_sz = PyDict_Size(cmd_env);
+        if (subprocess_env_sz != 0) {
+
+            if (!(subprocess_env = (char **)_malloc(sizeof(*subprocess_env) * (subprocess_env_sz + 1)))) {
+                PyErr_SetString(PyExc_MemoryError, "unable to allocate memory, bailing out");
+                goto cleanup;
+            }
+
+            for (i = 0; i <= subprocess_env_sz; i++) {
+                subprocess_env[i] = NULL;
+            }
+
+            Py_ssize_t pos = 0;
+            PyObject *key = NULL, *value = NULL;
+            for (i = 0; i < subprocess_env_sz && PyDict_Next(cmd_env, &pos, &key, &value); i++) {
+
+                char *env_key = as_string(key);
+                if (env_key == NULL) {
+                    PyErr_SetString(PyExc_TypeError, "env key is not a string");
+                    goto cleanup;
+                }
+
+                char *env_value = as_string(value);
+                if (env_value == NULL) {
+                    PyErr_SetString(PyExc_TypeError, "env value is not a string");
+                    _free(env_key);
+                    goto cleanup;
+                }
+
+                char *env = (char *)_malloc(strlen(env_key) + 1 + strlen(env_value) + 1);
```

Comment:
I know we're allocating char could we sizeof `*env` too. Just in case we change the type in the future for any reasons.

Reply:
Do you mean something like this:
```suggestion
                char *env = (char *)_malloc((strlen(env_key) + 1 + strlen(env_value) + 1) * sizeof(char));
```
?

so that, in case we migrate from `char` to `wchar_t` in the future, the required changes will be more obvious.

Reply:
More something like:

```c
char *env = (char *)_malloc((strlen(env_key) + 1 + strlen(env_value) + 1) * sizeof(*env));
```

That way we only have 1 type to change if we want to use `wchar_t` for example.

Reply:
I’ve just amended this PR to include your proposition: https://github.com/DataDog/datadog-agent/pull/5987/commits/a26e5b1a9119caf606364ecdaa899b06b100bb3f

Note however that, in case we want to migrate to `wchar_t` for example, all the string functions will need to be changed (`strlen` → `wcslen`, `strcpy` → `wcscpy`, `strcat` → `wcscat`, etc.),
So, rather than anticipating an hypothetical migration to wide characters, I think we should privilege code readability.

`malloc` expects a size in bytes whereas `strlen` returns a number of characters. So, multiplying by the size in bytes of a character perfectly makes sense from a unit perspective.